### ISSUE TITLE
Typescript import build error

### ISSUE THREAD

**Issue by yd021976** [feathersjs/feathers-client#179]
### Steps to reproduce
Packages managed with NPM and code written in typescript (développement with Aurelia JS framework).

The following code fall into following build error : 
> error TS2349 : Cannot invoke an expression whose type lacks a call signature. Type 'typeof "/Volumes/LaCie/www/aurelia-test2/node_modules/feathers-client/index"' has no compatible call signatures.

```
import * as io from 'socket.io-client';
import * as feathers from 'feathers-client';

let baseUrl = 'http://localhost:3030';
let socket = io.connect(baseUrl);

let client = feathers().configure(feathers.socketio(socket));
let messages = client.service('messages');

export { client, messages };
```

with this code alias "feathers" contains reference to interfaces FeathersApp and FeathersService. So, it's impossible to use `feathers()` or `feathers.default()` and impossible to use the client API (interfaces cannot be instantiated nor implemented in classes)

### Expected behavior
import "feathers" should provide a function or object to use feathers client API.

### Actual behavior
import "feathers" provide interfaces FeathersApp and FeathersService.
Seems it is due to "typings": "index.d.ts" sets in package.json
--> When file "index.d.ts" is removed, importing feathers-client seems work OK and no more build errors (but no intellisense/completion available in Sublime text editor)

### System configuration

Tell us about the applicable parts of your setup.
** Editor : Sublime Text

** Typescript framework : AureliaJS

**Module versions** (especially the part that's not working): V2.2.0

**NodeJS version**:

**Operating System**: MacOS 10.12.5

**Browser Version**: Safari 10.1.1

**React Native Version**: Not used

**Module Loader**: RequireJS

**Comment by eddyystop** [comment]
"Seems it is due to "typings": "index.d.ts" TypeScript defs are maintained by Feathers TS community.

A PR is welcome.

**Comment by yd021976** [comment]
What do you mean about "a PR is welcome" ?

**Comment by marshallswain** [comment]
@yd021976 it means that because nobody on the core team uses TypeScript, none of us knows how to fix it.  We rely on the TypeScript community to fix the issues.  So we would love to see a pull request (PR) that solves the problem.  Otherwise, maybe ask around in Slack to do some research on how to fix the issue?  I keep meaning to dive into TypeScript, and I will... someday.  ;)  Thanks for the great bug report.

**Comment by daffl** [comment]
Closed via #194
